intel-sgx/fortanix-sgx-tools/src/bin/ftxsgx-runner.rs

```diff
@@ -47,6 +48,30 @@ fn catch_sigbus() {
     }
 }
 
+#[cfg(unix)]
+fn catch_sigusr1() {
+    unsafe {
+        extern "C" fn handle_sigusr1(_signo: c_int, _info: *mut siginfo_t, _context: *mut c_void) {
+            eprintln!("SIGUSR1 triggered, thread_id: {:?}", std::thread::current().id());
+            let instruction_ptr = unsafe { (*(_context as *mut ucontext_t)).uc_mcontext.gregs[REG_RIP as usize] as *const u8};
+            // enclu instruction code
+            const ENCLU: [u8; 3] = [0x0f, 0x01, 0xd7];
+            let is_enclu = ENCLU.iter().enumerate().all(|(idx, v)| {
+                unsafe { *instruction_ptr.offset(idx as isize) == *v }
+            });
+            if is_enclu {
+                // Interrupt enclave execution by setting IP to the instruction following the ENCLU to mimic normal ENCLU[EXIT])
+                unsafe { (*(_context as *mut ucontext_t)).uc_mcontext.gregs[REG_RIP as usize] += 3 }
+            }
+            let _ = stderr().flush();
+        }
+        
+        let hdl = signal::SigHandler::SigAction(handle_sigusr1);
+        let sig_action = signal::SigAction::new(hdl, signal::SaFlags::empty(), signal::SigSet::empty());
+        signal::sigaction(signal::SIGUSR1, &sig_action).unwrap();
```

Comment:
USR1 is ok, but isn't e.g. HUP more appropriate? Combined with only conditionally rewriting the IP.

Reply:
Changed to HUP.
Calling this handler is now conditional for vdso/non-vdso. Is that what you mean by conditional rewriting IP?